lib/scope.js

```diff
@@ -73,6 +73,7 @@ ScopeDefinition.prototype.related = function(receiver, scopeParams, condOrRefres
     // Return from cache
     cb(null, self.__cachedRelations[name]);
   }
+  return cb.promise;
```

Comment:
Again, `related()` should not support promise API, thus `cb.promise` is never set and this change can be reverted.